tests/fixtures/root-paths-cases.json

```diff
@@ -0,0 +1,30 @@
+[
+  {
+    "root": "localhost",
+    "basePath": "http://localhost:80"
+  },
+  {
+    "root": "127.0.0.1",
+    "basePath": "http://127.0.0.1:80"
+  },
+  {
+    "root": "localhost:1234",
+    "basePath": "http://localhost:1234"
+  },
+  {
+    "root": "127.0.0.1:1234",
+    "basePath": "http://127.0.0.1:1234"
+  },
+  {
+    "root": "localhost:1234/url",
+    "basePath": "http://localhost:1234/url"
+  },
+  {
+    "root": "127.0.0.1:1234/url",
+    "basePath": "http://127.0.0.1:1234/url"
+  },
+  {
+    "root": "foo.com/api",
+    "basePath": "http://foo.com:80/api"
+  }
+]
```

Comment:
maybe add two https testcase, one with port, one without? Otherwise looks good so far 👍

Reply:
ping @Overdrivr ☝️🙏

Reply:
I have added a few tests with protocols like you requested.

Also added a test to check that if a root url has a trailing backslash (like `foo.com/`) `nock`ing will still work.

And finally I have added actual checks that node can intercept a request on those root urls. Currently it cannot.